### PR TITLE
Adding junit tests for Json creation

### PR DESCRIPTION
**Pull request by rmohan20**
These unit tests focus only on the structure of JSON. The logic behind processing XML is not covered here.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -76,7 +76,7 @@
   },
   "repository": "forcedotcom/sfdx-scanner",
   "scripts": {
-    "build": "./gradlew build",
+    "build": "./gradlew build test",
```
Comment (michaelhoefer) [inline package.json]: ?  We should not intermingle building with running tests here.  In the CI for example, we run build, then we run test, then we run the linter.

Instead, consider adding it to the test command below:

 "test": "./gradlew test && nyc mocha './test/**/*.test.ts'"

File: pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/catalog/PmdCatalogRule.java [hunk 2]
```diff
@@ -23,15 +34,15 @@
...
+    this.name = element.getAttribute(ATTR_NAME);
+    this.message = element.getAttribute(ATTR_MESSAGE);
     this.language = language;
     this.category = category;
     this.description = getDescription(element);
   }
 
   String getFullName() {
-    return getCategoryPath() + "/" + getName();
+    return getCategoryPath() + PATH_JOINER + getName();
```
Comment (michaelhoefer) [inline pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/catalog/PmdCatalogRule.java]: Traditionally (in java) this is called a path separator.  But wondering why you made it a constant?

File: package.json (hunk 1 above)
Comment (rmohan20) [reply package.json]: Good call. Will fix this.

File: pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/catalog/PmdCatalogRule.java (hunk 2 above)
Comment (rmohan20) [reply pmd-cataloger/src/main/java/sfdc/sfdx/scanner/pmd/catalog/PmdCatalogRule.java]: Had plans for adding a test for this, but ended up not doing that. Will change it back to what it was.

**Pull request merged**